pym/bob/scm/scm.py

```diff
@@ -41,15 +43,18 @@ def __eq__(self, other):
         return ((self.__match, self.__del, self.__set, self.__replace) ==
             (other.__match, other.__del, other.__set, other.__replace))
 
-    def mangle(self, scm):
+    def mangle(self, scm, env=Env()):
+        override = copy.deepcopy(self)
```

Comment:
Quite frankly I don't like the approach. As this is in some of the most performance critical paths of Bob I would prefer a more "classic" approach. :wink: 

```python
if self.__doesMatch(scm, env):
    rm, set, replace = self.__applyEnv(env)
    for d in rm:
...
```

The `__doesMatch` method could substitute the value only if the key is present. That may save some cycles if a key is missing. The `__applyEnv` method should just return a tuple of the substituted fields instead of mutating an object copy.